src/Components/LifecycleModalWindow/LifecycleModalWindow.tsx

```diff
@@ -0,0 +1,227 @@
+- isModalOpen - value managing if modal is open in UI, React.UseState()
+- handleModalToggle - function for handling close/open modal window, _event: React.MouseEvent | React.KeyboardEvent
+*/
+interface ModalWindowProps {
+  name: String | undefined;
+  modalData: string[] | undefined;
+  setModalData: React.Dispatch<React.SetStateAction<string[] | undefined>>;
+  isModalOpen: boolean;
+  handleModalToggle: (_event: any) => void; // any because <Modal onClose> stops working with anything else
+}
+
+export const LifecycleModalWindow: React.FunctionComponent<
+  ModalWindowProps
+> = ({ name, modalData, setModalData, isModalOpen, handleModalToggle }) => {
+  const [modalDataFiltered, setModalDataFiltered] = React.useState<
+    string[] | undefined
+  >();
+  const [activeSortIndex, setActiveSortIndex] = React.useState<
+    number | undefined
+  >();
+  const [activeSortDirection, setActiveSortDirection] =
+    React.useState<SortByDirection>();
+  const [inputValue, setInputValue] = React.useState("");
+
+  // When the modal window is opened, update it with new data/default values
+  React.useEffect(() => {
+    setModalDataFiltered(modalData);
+    setActiveSortIndex(undefined);
+    setActiveSortDirection(undefined);
+    setInputValue("");
+  }, [isModalOpen]);
+
+  const renderModalWindow = () => {
+    return (
+      <Modal
+        variant={ModalVariant.small}
+        isOpen={isModalOpen}
+        onClose={handleModalToggle} // TODO: problem when handleModalToggle = (_event: React.MouseEvent | React.KeyboardEvent) this doesn't work
+        aria-labelledby="scrollable-modal-title"
+        aria-describedby="modal-box-body-scrollable"
+      >
+        <ModalHeader
+          title="Systems"
+          labelId="scrollable-modal-title"
+          description={`${name} is installed on these systems. Click on a system name to view system details in Inventory.`}
+        />
+        <ModalBody
+          tabIndex={0}
+          id="modal-box-body-scrollable"
+          aria-label="Scrollable modal content"
+        >
+          {renderModalWindowTable(modalDataFiltered)}
+        </ModalBody>
+        <ModalFooter></ModalFooter>
+      </Modal>
+    );
+  };
+
+  const renderModalWindowTable = (data: string[] | undefined) => {
+    if (data === undefined) {
+      return "";
+    }
+
+    return (
+      <div>
+        {renderFilterBoxModalWindow()}
+        <Table variant="compact">
+          <Thead>
+            <Tr>
+              <Th sort={getSortParamsModalWindow(0, data)}>Name</Th>
+            </Tr>
+          </Thead>
+          <Tbody>
+            {data?.map((item, index) => (
+              <Tr key={index}>
+                <Td dataLabel="Name">
+                  <Button variant="link">{item}</Button>
+                </Td>
+              </Tr>
+            ))}
+          </Tbody>
+        </Table>
+      </div>
+    );
+  };
+
+  const renderFilterBoxModalWindow = () => {
+    return (
+      <TextInputGroup>
```

Comment:
```suggestion
      <TextInputGroup style={{ maxWidth: '140px' }}>
```

Reply:
I believe using the `.scss` files for formatting would be better instead of the inline ones, but... This works, and the styles in the file don't. Maybe something to change later, but for now totally fine :D

Thank you for looking at it and make the modal prettier!